core/src/avm2/globals/flash/geom/perspective_projection.rs

```diff
@@ -0,0 +1,67 @@
+use std::f64::consts::PI;
+
+use crate::avm2::globals::slots::flash_geom_perspective_projection as pp_slots;
+use crate::avm2::{Activation, Error, Object, TObject, Value};
+use crate::display_object::TDisplayObject;
+use crate::{avm2_stub_getter, avm2_stub_setter};
+
+const DEG2RAD: f64 = PI / 180.0;
+
+fn get_width<'gc>(activation: &mut Activation<'_, 'gc>, this: Object<'gc>) -> f64 {
+    let dobj = this
+        .get_slot(pp_slots::DISPLAY_OBJECT)
+        .as_object()
+        .and_then(|e| e.as_display_object());
+
+    match dobj {
+        // Not associated with any DO
+        None => 500.0,
+        // Stage's PerspectiveProjection
+        Some(dobj) if dobj.as_stage().is_some() => 500.0,
```

Comment:
I don't know where those values come from. It's the output from the test result in FP.